### PR TITLE
Add UserDefinedNetwork resource using class generator

### PR DESCRIPTION
**Pull request by sbahar619**
##### Short description:
This PR introduces the UserDefinedNetwork class, which supports the creation and management of UserDefinedNetwork custom resources (CRs) in a Kubernetes environment.
##### What this PR does / why we need it:
To support this epic:
https://issues.redhat.com/browse/CNV-45524
## Summary by CodeRabbit

- **New Features**
	- Introduced a new constant `K8S_OVN_ORG` for Kubernetes API group.
	- Added the `UserDefinedNetwork` class for flexible network configuration in OpenShift, supporting various topologies and configurations.
- **Bug Fixes**
	- Implemented error handling in the `to_dict` method to ensure required arguments are provided.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -296,6 +296,7 @@ class ApiGroup:
         K8S_CNI_CNCF_IO: str = "k8s.cni.cncf.io"
         K8S_MARIADB_COM: str = "k8s.mariadb.com"
         K8S_V1_CNI_CNCF_IO: str = "k8s.v1.cni.cncf.io"
+        K8S_OVN_ORG: str = "k8s.ovn.org/v1"
```
Comment (rnetser) [inline ocp_resources/resource.py:299]: please move before `K8S_OVN_ORG`

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py (hunk 1 above)
Comment (rnetser) [reply ocp_resources/resource.py:299]: please address

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/user_defined_network.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    """
+
+    api_group: str = NamespacedResource.ApiGroup.K8S_OVN_ORG
+
+    def __init__(
+        self,
+        layer2: Optional[Dict[str, Any]] = None,
+        layer3: Optional[Dict[str, Any]] = None,
+        local_net: Optional[Dict[str, Any]] = None,
+        topology: Optional[str] = "",
```
[the remark this answers is not in the document]
Comment (sbahar619) [reply ocp_resources/user_defined_network.py:19]: Not touching it to save class generator automatic version.

File: ocp_resources/resource.py (hunk 1 above)
Comment (sbahar619) [reply ocp_resources/resource.py:298]: Done.

**Review by rnetser — APPROVED** [review]

**Review by yossisegev — APPROVED** [review]
/lgtm

**Review by myakove — APPROVED** [review]

**Pull request merged**